### PR TITLE
Dental Implant surgery no longer restarts itself for every pill you implant

### PR DESCRIPTION
**Pull request by MGOOOOOO**
## About The Pull Request

Currently, the "Dental Implant" surgery requires you to restart the operation after implanting only one pill, which is an annoying inconvenience. 

Implanting 5 pills before PR:

https://github.com/tgstation/tgstation/assets/97645027/d9bb74b6-7c7f-4b62-8b3d-4fe4b616dc2e

This PR aims to fix this problem by making the "Implant pill" portion of the surgery repeatable by introducing a new step. After you implant a pill, you can either search their mouth with your hands for another tooth to implant, or use a cautery to finalize the surgery!

Implanting 5 pills after PR:

https://github.com/tgstation/tgstation/assets/97645027/ead38fb5-8ddc-463a-80c5-089434950646

## Why It's Good For The Game

If you're going through the trouble with another person to implant pills into your teeth, you'll likely want more then one pill implanted. Aside from making the surgery more consistent with other repeatable surgeries, this makes the entire process less tedious to go through for the surgeon.

## Changelog
:cl:
qol: The "Dental Implant" surgery no longer forces itself to restart after implanting one pill. Now implanting pills in the surgery has another step to either search the patient's mouth for another tooth to implant, or cauterization to end the surgery. 
/:cl:

### CODE REVIEW

**Comment by MGOOOOOO** [comment]
WHOOPSIE, MADE A LITTLE ERROR. GIVE ME A MOMENT.......

**Comment by MGOOOOOO** [comment]
There we go. This is just a QOL PR, sorry!

**Comment by Wallemations** [comment]
This doesn't really make much sense to me tbh, is the idea that you're stuffing all those pills into a single tooth now instead of drilling 5 different teeth?

**Comment by AtomTheProphet** [comment]
Yeah. I agree it should be less tedious by restarting the surgery each time, but you should have to drill for each pill you're implanting.

**Comment by MGOOOOOO** [comment]
The flavor text just says you drill into the bone instead of teeth, but fair I can make it go back to the drill part after initial pill.

**Comment by Metekillot** [comment]
Seems more like a balance change than a QOL one.

**Comment by MGOOOOOO** [comment]
Alright, I've changed the surgery to have an extra step. After implanting a pill, you have two options - Either search their teeth with your hands for another tooth to implant, or use a cautery to complete the surgery. I'll be updating the PR body with the new info shortly!

File: code/modules/surgery/dental_implant.dm [hunk 1]
```diff
@@ -64,3 +68,32 @@
 		item_target.reagents.trans_to(owner, item_target.reagents.total_volume, transferred_by = owner, methods = INGEST)
 	qdel(target)
 	return TRUE
+
+/datum/surgery_step/search_teeth
+	name = "search teeth (hand)"
+	accept_hand = TRUE
+	time = 20
```
Comment (MrMelbert) [inline code/modules/surgery/dental_implant.dm:75]:
```suggestion
	time = 2 SECONDS
```

Comment (MGOOOOOO) [reply code/modules/surgery/dental_implant.dm:75]: Ah right, thanks!

**Pull request merged**